### PR TITLE
Consider any azp claim when validating aud.

### PR DESCRIPTION
**Pull request by petea**
Addresses issue #418

### CODE REVIEW

File: Source/OIDAuthorizationService.m [hunk 1]
```diff
@@ -575,9 +575,11 @@ + (void)performTokenRequest:(OIDTokenRequest *)request
       }
 
       // OpenID Connect Core Section 3.1.3.7. rule #3
-      // Validates that the audience of the ID Token matches the client ID.
+      // Validates that the audience of the ID Token matches the client ID unless a matching azp
+      // Claim is present.
```
Comment (WilliamDenniss) [inline Source/OIDAuthorizationService.m]: We reference `OpenID Connect Core Section 3.1.3.7. rule #3` for the first part of the rule. Can you add a sectional reference for the second part (i.e. the `azp` section)?

File: Source/OIDAuthorizationService.m [hunk 2]
```diff
@@ -574,10 +574,12 @@ + (void)performTokenRequest:(OIDTokenRequest *)request
         return;
       }
 
-      // OpenID Connect Core Section 3.1.3.7. rule #3
-      // Validates that the audience of the ID Token matches the client ID.
+      // OpenID Connect Core Section 3.1.3.7. rule #3 & Section 2 azp Claim
+      // Validates that the audience of the ID Token matches the client ID unless a matching azp
```
Comment (WilliamDenniss) [inline Source/OIDAuthorizationService.m]: nit, I might phrase it as
"Validates that the aud (audience) Claim contains the client ID, or that the azp (authorized party) Claim matches the client ID."

Comment (petea) [reply Source/OIDAuthorizationService.m]: Yes, that's better.

**Comment by WilliamDenniss** [comment]
Merging based on my [review ](https://github.com/openid/AppAuth-iOS/issues/418#issuecomment-519634483) of the documented `azp` functionality.

**Comment by crowe2134** [comment]
cf5b15a6e02ee509c5966c5dacbc45a0bef7c5cf

File: Source/OIDAuthorizationService.m (hunk 1 above)
Comment (crowe2134) [reply Source/OIDAuthorizationService.m]: [object Object]

Comment (crowe2134) [reply Source/OIDAuthorizationService.m]: [object Object]

**Pull request merged**